src/test/java/org/opentripplanner/updater/trip/TimetableSnapshotSourceTest.java

```diff
@@ -574,6 +574,45 @@ public void scheduled() {
       assertEquals(90, originalTripTimesForToday.getDepartureDelay(2));
     }
 
+    /**
+     * This test just asserts that trip with start date that is outside the service period doesn't
+     * throw an exception and is ignored instead.
+     */
+    @Test
+    public void invalidTripDate() {
+      // GIVEN
+
+      String scheduledTripId = "1.1";
+
+      var serviceDateOutsideService = SERVICE_DATE.minusYears(10);
+      var builder = new TripUpdateBuilder(
+        scheduledTripId,
+        serviceDateOutsideService,
+        SCHEDULED,
+        transitModel.getTimeZone()
+      )
+        .addDelayedStopTime(1, 0)
+        .addDelayedStopTime(2, 60, 80)
+        .addDelayedStopTime(3, 90, 90);
+
+      var tripUpdate = builder.build();
+
+      var updater = defaultUpdater();
+
+      // WHEN
+      updater.applyTripUpdates(
```

Comment:
I know that not many tests in this file do this yet but please assert that the `UpdateResult` actually contains the error we expect. Otherwise I think this PR is good to merge.

Reply:
Good point, updated in https://github.com/opentripplanner/OpenTripPlanner/pull/4861/commits/8f0be0552beb909bf0cd80c2f49ad2537d825c26